rocAL/rocAL/include/decoders/image/decoder.h

```diff
@@ -46,6 +46,12 @@ class DecoderConfig
     virtual DecoderType type() {return _type; };
     DecoderType _type = DecoderType::TURBO_JPEG;
     std::vector<Parameter<float>*> _crop_param;
+    void set_random_area(std::vector<double> &random_area) { _random_area = std::move(random_area); }
```

Comment:
arguments are passed by reference and copied to new. Better to use a pointer instead of copying to new vector.